app/scripts/modules/core/pipeline/config/stages/bake/bakeStageChooseOs.component.ts

```diff
@@ -0,0 +1,39 @@
+import {IComponentController, IComponentOptions, module} from 'angular';
+
+export interface IBaseOsOption {
+  id: string;
+  shortDescription?: string;
+  detailedDescription: string;
+  isImageFamily?: boolean;
+}
+
+export class BakeStageChooseOSController implements IComponentController {
+
+  public model: any;
+  public baseOsOptions: IBaseOsOption[];
+
+  public getBaseOsDescription(baseOsOption: IBaseOsOption): String {
+    return baseOsOption.id + (baseOsOption.shortDescription ? ' (' + baseOsOption.shortDescription + ')' : '');
+  }
+
+  public getBaseOsDetailedDescription(baseOsOption: IBaseOsOption): String {
+    return baseOsOption.detailedDescription + (baseOsOption.isImageFamily ? ' (family)' : '');
+  }
+
+  public showRadioButtons(): boolean {
+    return this.baseOsOptions.length <= 2
+  }
```

Comment:
Let's move this to a private property on the controller and put it into an `$onInit` method:

```typescript
...
public baseOsOptions: IBaseOsOption[];

private showRadioButtons = false;

public $onInit(): void {
  this.showRadioButtons = this.baseOsOptions.length <= 2;
}
...
```
When you put a method into an `ng-if` or `ng-repeat`, Angular will call it every time there's a user interaction. On the pipeline config page, it's not too bad - especially here, since the method is very fast - but it's a pattern we try to avoid.